db/migrate/20150629180552_change_password_type_user.rb

```diff
@@ -0,0 +1,6 @@
+class ChangePasswordTypeUser < ActiveRecord::Migration
+  def change
+    remove_column :users, :password
+    add_column :users, :password, :varchar, null: false
```

Comment:
Two. Add password_hash column in the first migration, hash the user's password into it, change and test the system to use the new column, and if everything is fine create a new migration to drop the password column. There would be no downtime for this.